### PR TITLE
SegmentedControl: Add Density Variants

### PR DESCRIPTION
**Pull request by rlingineni**
### Summary

Add 'sm', 'md', 'large' to density variants

#### What changed?
Before/After:

image

From a high level, what are the changes this PR introduces? (No need to recount line-by-line, we can see that.)

### Links

- [Jira](https://jira.pinadmin.com/browse/GESTALT-XXXX)
- [TDD](link to Paper doc)
- [Figma](link to Figma file)

### Checklist

- [ ] Added unit and Flow Tests
- [ ] Added documentation + accessibility tests
- [ ] Verified accessibility: keyboard & screen reader interaction
- [ ] Checked dark mode, responsiveness, and right-to-left support
- [ ] Checked stakeholder feedback (e.g. Gestalt designers)

### CODE REVIEW

**Comment by AlbertCarreras** [comment]
The small variant seems to present different border radius

**Comment by hectoid** [comment]
@rlingineni sorry, that's a mistake on my end. The auto radius should be 8px.

**Comment by hectoid** [comment]
@rlingineni in the spec, the outer gap is also 4px. I still see it as 1px in the examples. Figma: 
image

**Comment by rlingineni** [comment]
Fixed. @hectoid - is devexample mode enabled?

https://deploy-preview-3275--gestalt.netlify.app/web/segmentedcontrol?devexample=true#Sizes

**Comment by hectoid** [comment]
@rlingineni oh, I forget about that. I see it now in dev mode. looks good! separate issue, I'm also seeing these wrap. Can the box be made wider so they don't wrap? 
image

**Comment by rlingineni** [comment]
@hectoid I noticed they wrap in devmode, and but don't in prod. Guessing it's dev mode wonkiness. So I didn't touch them.

File: packages/gestalt/src/SegmentedControl.js [hunk 1]
```diff
@@ -30,27 +30,44 @@ type Props = {
    * Index of element in `items` that is currently selected.
    */
   selectedItemIndex: number,
+  /**
+   * Size of the Segmented Control. Default is `md`.
```
Comment (AlbertCarreras) [inline packages/gestalt/src/SegmentedControl.js:34]: nit. We dont usually specify the default in the descrption. The table has it.

File: packages/gestalt/src/SegmentedControl.js [hunk 2]
```diff
@@ -13,7 +13,7 @@ type OnChange = ({
   activeIndex: number,
 }) => void;
 
-type Props = {
+type Props = {|
```
Comment (AlbertCarreras) [inline packages/gestalt/src/SegmentedControl.js:16]: We don't need strict types. they were removed

try running yarn if it complains about it. See https://github.com/pinterest/gestalt/pull/3254

File: packages/gestalt/src/SegmentedControl.js [hunk 3]
```diff
@@ -30,27 +30,44 @@ type Props = {
...
-}) {
+  size: 'sm' | 'md' | 'lg',
+|}) {
   const { isFocusVisible } = useFocusVisible();
-  const cs = classnames(styles.item, focusStyles.hideOutline, {
-    [styles.itemIsNotSelected]: !isSelected,
-    [styles.itemIsSelected]: isSelected,
-    [focusStyles.accessibilityOutline]: isFocusVisible,
-  });
+  const cs = classnames(
```
Comment (AlbertCarreras) [inline packages/gestalt/src/SegmentedControl.js:61]: can we inline this?

**Review by AlbertCarreras — APPROVED** [review]
R+ with comments

**Comment by hectoid** [comment]
@rlingineni suggest better labels for these to make the info easier to digest so we aren't just repeating the sizes:
image

Instead: All | Published | Drafts

**Comment by hectoid** [comment]
@rlingineni font size for Lg and Md should be 300; they have 200.
image

For sizes, we should include an example with an icon. I can't tell if icons are the correct size + would be good for people to see what that looks like at different densities.

**Comment by hectoid** [comment]
sorry @AlbertCarreras wasn't sure how to be sure to wait for design review before this gets merged. I'm finding some issues, but I don't have approval/reviewer permissions.

**Comment by hectoid** [comment]
@rlingineni those are the only things I found: font size, and nits on labeling examples. Looks good otherwise. Thanks!

File: packages/gestalt/src/SegmentedControl.js [hunk 4]
```diff
@@ -30,27 +30,44 @@ type Props = {
    * Index of element in `items` that is currently selected.
    */
   selectedItemIndex: number,
+  /**
+   * Size of the Segmented Control. Default is `md`.
+   */
+  size?: 'sm' | 'md' | 'lg',
+|};
+
+const applyDensityStyle = (s: 'sm' | 'md' | 'lg') => styles[`${s}`];
```
[the remark this answers is not in the document]
Comment (rlingineni) [reply packages/gestalt/src/SegmentedControl.js:39]: done

File: playwright/visual-test/SegmentedControl-dark.spec.mjs-snapshots/SegmentedControl-dark-chromium-darwin.png
[the remark this answers is not in the document]
Comment (rlingineni) [reply playwright/visual-test/SegmentedControl-dark.spec.mjs-snapshots/SegmentedControl-dark-chromium-darwin.png:1]: done

File: packages/gestalt/src/SegmentedControl.css [hunk 5]
```diff
@@ -16,7 +29,21 @@
   border-radius: var(--rounding-300);
```
[the remark this answers is not in the document]
Comment (rlingineni) [reply packages/gestalt/src/SegmentedControl.css:29]: Yeah, not needed

File: docs/pages/web/segmentedcontrol.js [hunk 6]
```diff
@@ -47,27 +48,34 @@ export default function DocsPage({ generatedDocGen }: { generatedDocGen: DocGen
...
           description="Segmented Control can have responsive widths where the width of an item is based on its content."
+          title="Responsive"
         >
           <MainSection.Card
             cardSize="lg"
             sandpackExample={<SandpackExample name="Responsive Example" code={responsiveExample} />}
           />
         </MainSection.Subsection>
+        <MainSection.Subsection
+          description="Segmented Control can have different sizes. The default size is medium (40px). Large is 48px. For a dense variant, use the `sm` (32px) size. "
```
[the remark this answers is not in the document]
Comment (rlingineni) [reply docs/pages/web/segmentedcontrol.js:71]: fixed

**Comment by rlingineni** [comment]
@hectoid - can you take a look when you get a chance. I added icon examples, and tweaked the fontSizes. For the Icons, I noticed that the sizes are non-semantic. 

https://gestalt.pinterest.systems/web/icon

In this case, we're using a size 20 icon, but that's actually not pre-defined. It's nbd, but maybe we can align icon sizes in the future to our density scale

**Comment by rlingineni** [comment]
@hectoid - can I get a 👍🏾 on this as well

**Comment by hectoid** [comment]
👍🏼

**Comment by hectoid** [comment]
@rlingineni approved!

**Pull request merged**